sessiontxn/internal/txn.go

```diff
@@ -62,9 +61,11 @@ func CommitBeforeEnterNewTxn(ctx context.Context, sctx sessionctx.Context) error
 }
 
 // GetSnapshotWithTS returns a snapshot with ts.
-func GetSnapshotWithTS(s sessionctx.Context, ts uint64) kv.Snapshot {
+func GetSnapshotWithTS(s sessionctx.Context, ts uint64, interceptor kv.SnapshotInterceptor) kv.Snapshot {
 	snap := s.GetStore().GetSnapshot(kv.Version{Ver: ts})
-	snap.SetOption(kv.SnapInterceptor, temptable.SessionSnapshotInterceptor(s))
+	if interceptor != nil {
+		snap.SetOption(kv.SnapInterceptor, interceptor)
```

Comment:
The old implementation always set the SnapInterceptor. What if interceptor is nil?

Reply:
To make the code strong. Actually, interceptor is not nil for the current implement.